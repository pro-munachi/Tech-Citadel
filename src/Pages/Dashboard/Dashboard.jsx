import { Box, Flex, Text } from "@chakra-ui/react";
import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";
import Navbar from "../../components/Navbar/Navbar";
import Student from "../../components/Student/Student";
import Tiles from "../../components/Tiles/Tiles";
import { dashboardInfo } from "../../state/actions/dashboardAction";

const Dashboard = () => {
  const dispatch = useDispatch();

  const navigate = useNavigate();

  const state = useSelector((state) => state.login);
  const { userInfo } = state;

  const dash = useSelector((state) => state.dashboard);
  const { data } = dash;

  const user = JSON.parse(localStorage.getItem("userInformations"));

  React.useEffect(() => {
    getData();
    if (!user) {
      navigate("/login");
    }
  }, []);

  const getData = () => {
    dispatch(dashboardInfo(userInfo?.uid || user?.uid));
  };
  return (
    <Box>
      <Navbar />
      <Box width="80%" mx="auto" pb="40px">
        <Student data={data} />

        <Box mt="40px">
          <Text fontSize={["25px", "25px", "30px", "30px"]} textAlign="center">
            Subjects
          </Text>
          <Flex flexWrap="wrap" justifyContent="space-between">
            {data?.subjects?.map((data, i) => (
              <Box key={i} mt="45px">
                <Tiles data={data} key={i} />
              </Box>
            ))}
          </Flex>
        </Box>
      </Box>
    </Box>
  );
};

export default Dashboard;
