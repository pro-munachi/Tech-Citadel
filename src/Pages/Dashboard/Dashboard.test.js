import { render } from "@testing-library/react";
import Dashboard from "./Dashboard";
import { BrowserRouter } from "react-router-dom";
import { ChakraProvider, theme } from "@chakra-ui/react";
import store from "../../state/store";
import { Provider } from "react-redux";

const Wrapper = ({ children }) => {
  <ChakraProvider theme={theme}>
    <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  </ChakraProvider>;
};

test("Test Dashboard", async () => {
  const { asFragment } = render(
    <Wrapper>
      <Dashboard />
    </Wrapper>
  );

  expect(asFragment()).toMatchSnapshot();
});
