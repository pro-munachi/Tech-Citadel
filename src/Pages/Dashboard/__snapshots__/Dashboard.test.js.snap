// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard 1`] = `<DocumentFragment />`;
